src/citrine/resources/ara_job.py

```diff
@@ -0,0 +1,179 @@
+                  "failure_reason": value.failure_reason, "dependencies": value.dependencies}
+        return result
+
+    def _deserialize(self, value: dict) -> TaskNode:
+        return TaskNode.build(value)
+
+
+class JobStatusResponse(Resource['JobStatusResponse']):
+    """
+    [ALPHA] a response to a job status check
+
+    Parameters
+    ----------
+    job_type: str
+        the type of job for this status report
+    status: str
+        the actual status of the job
+    tasks: List[TaskNode]
+        all of the constituent task required to complete this job
+    """
+
+    job_type = properties.String("job_type")
+    status = properties.String("status")
+    tasks = properties.List(TaskNodeType, "tasks")
+
+    def __init__(
+            self,
+            job_type: str,
+            status: str,
+            tasks: List[TaskNode]
+    ):
+        self.job_type = job_type
+        self.status = status
+        self.tasks = tasks
+
+
+class AraJobFramework(Resource["AraJobFramework"]):
+    """
+    [ALPHA] a resource for submitting a job and getting job status
+
+    Parameters
+    ----------
+    session: Session
+        the remote session for hitting framework endpoints
+    """
+    _path_template = 'projects/{project_id}'
+
+    def __init__(self, session: Session):
+        self.session = session
+
+    def build_ara_table(self, project: Project,
+                        ara_def: AraDefinition) -> Union[ApiError, AraJobSubmissionResponse]:
+        """
+        [ALPHA] submit an ara table construction job
+
+        This method makes a call out to the job framework to start a new job to build
+        the respective table for a given AraDefinition.
+
+        Parameters
+        ----------
+        project: Project
+            The project on behalf of which the job executes.
+        ara_def: AraDefinition
+            the ara definition describing the new table
+        """
+        job_id = uuid4()
+        url_suffix: str = "/ara-definitions{ara_definition}/versions/{version_number}/build?job_id={job_id}"
+        if ara_def.version_number is None:
+            return ApiError(400, "Version number was not present request", None)
+        elif ara_def.definition_uid is None:
+            return ApiError(400, "Ara Definition UID was not present request", None)
+        elif project.uid is None:
+            return ApiError(400, "Project UUID was not present request", None)
+        else:
+            path: str = self._path_template.format(
+                project_id=project.uid
+            ) + url_suffix.format(
+                ara_definition=ara_def.definition_uid,
+                version_number=ara_def.version_number,
+                job_id=job_id
+            )
+            response: dict = self.session.post_resource(path=path)
+            return AraJobSubmissionResponse.build(response)
+
+    def get_job_status(self, project: Project, job_id: str):
```

Comment:
This is probably fine for now, but we will at least also want this on a more generic job collection. Job status has no ara-specific requirements and can be checked regardless of the endpoint which launched it.

Reply:
On the surface this feels similar to the flow on the Orion side of the client--the user submits requests that take a long time and hence needs to be able to get an updated job status. On the backend the mechanics might be very different, but client-side can we use the same framework for both?